platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/ConfigurationFileProxy.java

```diff
@@ -69,46 +75,29 @@ public ConfigurationFileProxy(ConfigurationStore configurationStore) {
         }
     }
 
-    protected BundleContext getContext() {
-        Bundle cxfBundle = FrameworkUtil.getBundle(ConfigurationFileProxy.class);
-        if (cxfBundle != null) {
-            return cxfBundle.getBundleContext();
-        }
-        return null;
+    public void writeSolrConfiguration(String core) {
+        writeSolrConfiguration(Paths.get(this.dataDirectory.getAbsolutePath(), core, "conf")
+                .toFile());
     }
 
     /**
-     * Writes the solr configuration files out of the bundle onto the disk. This method requires
-     * that the dataDirectoryPath has been set. If the code is run in an OSGi container, it will
-     * automatically have a default dataDirectory location set and will not require setting
-     * dataDirectory ahead of time.
+     * Writes the solr configuration files out of the classpath onto the disk.
      */
-    public void writeBundleFilesTo(File configDir) {
-        BundleContext bundleContext = getContext();
-        if (bundleContext != null && configDir != null) {
-            boolean directoriesMade = configDir.mkdirs();
-            LOGGER.debug("Solr Config directories made?  {}", directoriesMade);
-
-            @SuppressWarnings("rawtypes")
-            Enumeration entries = bundleContext.getBundle()
-                    .findEntries(SOLR_CONFIG_LOCATION_IN_BUNDLE, "*.*", false);
-
-            while (entries.hasMoreElements()) {
-                URL resourceURL = (URL) (entries.nextElement());
-                LOGGER.debug("Found {}", resourceURL);
-
-                try (InputStream inputStream = resourceURL.openStream()) {
-                    String fileName = FilenameUtils.getName(resourceURL.getPath());
-                    File currentFile = new File(configDir, fileName);
-
-                    if (!currentFile.exists()) {
-                        try (FileOutputStream outputStream = new FileOutputStream(currentFile)) {
-                            long byteCount = IOUtils.copyLarge(inputStream, outputStream);
-                            LOGGER.debug("Wrote out {} bytes.", byteCount);
-                        }
-                    }
+    public void writeSolrConfiguration(File configDir) {
+        boolean directoriesMade = configDir.mkdirs();
+        LOGGER.debug("Solr Config directories made?  {}", directoriesMade);
+
+        for (String filename : SOLR_CONFIG_FILES) {
+            File currentFile = new File(configDir, filename);
+            File backupFile = new File(configDir, filename + ".bak");
+            if (!currentFile.exists() && !backupFile.exists()) {
```

Comment:
Should we really skip the file if a backup already exists? If so, should we have two separate conditions and log different messages since their severity (can't copy the file at all vs. skipped because a backup file already existed) seem to not be the same.